src/Features/Core/Portable/UpgradeProject/AbstractUpgradeProjectCodeFixProvider.cs

```diff
@@ -39,15 +39,24 @@ protected ImmutableArray<CodeAction> GetUpgradeProjectCodeActionsAsync(CodeFixCo
             var project = context.Document.Project;
             var solution = project.Solution;
             var newVersion = SuggestedVersion(context.Diagnostics);
+
             var result = new List<CodeAction>();
             var language = project.Language;
 
+            var upgradeableProjects = solution.Projects.Where(p => CanUpgrade(p, language, newVersion)).AsImmutable();
+
+            if (upgradeableProjects.Length == 0)
+            {
+                return ImmutableArray<CodeAction>.Empty;
```

Comment:
so, I guess calling context.RegisterFixes with empty list is fine?

Reply:
you probably can use Skip(1).Any() but. I guess, not worth optimization.

Reply:
Yes, I believe this is the line that handles it in [CodeFixServices.cs](http://source.roslyn.io/#Microsoft.CodeAnalysis.EditorFeatures/Implementation/CodeFixes/CodeFixService.cs,397). It looks like null is handled earlier in the path, but I prefer empty array to null for clarity. The fixer decided explicitly it didn't need to provide fixes. Null isn't indicative of that knowledge